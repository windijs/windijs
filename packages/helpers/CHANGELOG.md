# @windijs/helpers

## 1.1.2

### Patch Changes

- [`39b28846`](https://github.com/windijs/windijs/commit/39b28846c780dc1c39ca30abfcd8e9c554aadb63) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: move dts generators to plugin-utils

## 1.1.1

### Patch Changes

- [`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: export package.json

* [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2) Thanks [@voorjaar](https://github.com/voorjaar)! - build: switch to es2018 for better tree shaking

* Updated dependencies [[`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3), [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2)]:
  - @windijs/shared@1.0.5

## 1.1.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

### Patch Changes

- [`8ab80baf`](https://github.com/windijs/windijs/commit/8ab80bafda9ab832d8d53f287e200f7fd497b7b6) Thanks [@voorjaar](https://github.com/voorjaar)! - chore: new versions

* [`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: remove circular devDependencies

- [`e9a948cd`](https://github.com/windijs/windijs/commit/e9a948cd44d003a2c1e91ea2927658f59e3746dc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new resetStyleMeta helper

* [`9940db66`](https://github.com/windijs/windijs/commit/9940db664965f9e7c04b8e6831cb035b79f2b212) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesDts with theme

- [`d0d14195`](https://github.com/windijs/windijs/commit/d0d14195bea0f846097a9738da82278ca97468bc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new handler types

* [`6a71ee76`](https://github.com/windijs/windijs/commit/6a71ee76737a9309b31b06adc33a695461d1583d) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: add dtsSetup and dtsHandler

- [`da5e3466`](https://github.com/windijs/windijs/commit/da5e3466071e7c780b3875e177dad87e14495d3c) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesJs with theme

* [`ba98151d`](https://github.com/windijs/windijs/commit/ba98151d7050fe6832999cf4c7ce0bec48d26fe0) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: nested StyleObject in ArrayHelper

* Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4

## 1.1.0-beta.5

### Patch Changes

- [`da5e3466`](https://github.com/windijs/windijs/commit/da5e3466071e7c780b3875e177dad87e14495d3c) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesJs with theme

* [`ba98151d`](https://github.com/windijs/windijs/commit/ba98151d7050fe6832999cf4c7ce0bec48d26fe0) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: nested StyleObject in ArrayHelper

## 1.1.0-beta.4

### Patch Changes

- [`9940db66`](https://github.com/windijs/windijs/commit/9940db664965f9e7c04b8e6831cb035b79f2b212) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesDts with theme

## 1.1.0-beta.3

### Patch Changes

- [`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: remove circular devDependencies

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/shared@1.0.4-beta.1

## 1.1.0-beta.2

### Patch Changes

- [`d0d14195`](https://github.com/windijs/windijs/commit/d0d14195bea0f846097a9738da82278ca97468bc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new handler types

* [`6a71ee76`](https://github.com/windijs/windijs/commit/6a71ee76737a9309b31b06adc33a695461d1583d) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: add dtsSetup and dtsHandler

* Updated dependencies [[`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4-beta.0

## 1.1.0-beta.1

### Patch Changes

- [`8ab80baf`](https://github.com/windijs/windijs/commit/8ab80bafda9ab832d8d53f287e200f7fd497b7b6) Thanks [@voorjaar](https://github.com/voorjaar)! - chore: new versions

* [`e9a948cd`](https://github.com/windijs/windijs/commit/e9a948cd44d003a2c1e91ea2927658f59e3746dc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new resetStyleMeta helper

## 1.1.0-beta.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

## 1.0.3

### Patch Changes

- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3

## 1.0.2

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/shared@1.0.2

## 1.0.1

### Patch Changes

- chore: release v1.0.1
- Updated dependencies
  - @windijs/shared@1.0.1

## 1.0.0
