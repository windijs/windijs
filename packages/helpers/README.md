# @windi/helpers

Operations related to css.

## CSS In JS

- $
- css
- prop
- atomic
- unify
- inline
- bundle
- injectCSS
- escapeCSS
- applyVariant
- cssInJsLoader
- useArrayHelper
- useStyleLoader

## Namer

- alphaNamer
- atomicNamer
- hashNamer
- useNamer
- nameStyle

## Meta

- getMeta
- getUid
- resetMeta
- pushMetaProp
- updateMetaType

## Utils

- isStyleObject
- getStyleVariants
- getStyleProps
- getStyleIdent
- baseStyleTarget
- baseStyleHandler
- getFirstVar
- useProxy
- setProxy
- isProxy

## Color Process

- digitToHEX
- rgbToHEX
- hexToRGB
- sliceColor
- rgbToHSL
- hueToRGB
- hslToRGB
- hwbToRGB
- adjustHue
- adjustSaturation
- adjustLightness
- adjustRed
- adjustGreen
- adjustBlue
- adjustAlpha
- mixColor
- subMixColor
- colorLuminance
- getLightColor
- getDarkColor
- Color
    - static hex
    - static rgb
    - static rgba
    - static hsl
    - static hsla
    - static hwb
    - static mix
    - static subcolormix
    - get hex
    - get rgb
    - get rgba
    - get hsl
    - get hsla
    - get hwb
    - get red
    - get blue
    - get green
    - get hue
    - get saturation
    - get lightness
    - get alpha
    - get opacity
    - get whiteness
    - get blacknness
    - get ieHexStr
    - invert()
    - adjustRed()
    - adjustGreen()
    - adjustBlue()
    - adjustHue()
    - adjustSaturation()
    - adjustLightness()
    - adjustAlpha()
    - complement()
    - saturate()
    - desaturate()
    - grayscale()
    - lighten()
    - darken()
    - opacify()
    - transparentize()
    - fadeIn()
    - fadeOut()
    - adjust()
    - scale()
    - change()
    - lightenSet()
    - darkenSet()
    - desaturateSet()
    - complementSet()
    - invertSet()

## CSS Units

- color
- percent
- deg, grad, rad, turn
- s, ms
- fr
- $in
- dpi, dpcm, dppx, x
- px, pc, pt, cm, mm, Q, ch, ex, em, rem, vw, vh, vmax, vmin

## CSS Functions

- filters
- transforms
- quote
- attr
- url
- $var
- path
- hwb
- dropShadow
- counters
- circle
- ellipse
- inset
- polygon
- matrix
- matrix3d
- perspective
- rotate
- rotate3d
- rotateX
- rotateY
- rotateZ
- scale
- scale3d
- scaleX
- scaleY
- scaleZ
- skew
- skewX
- skewY
- translate
- translate3d
- translateX
- translateY
- translateZ
- steps
- calc
- clamp
- max
- min
- abs
- sign
- blur
- brightness
- contrast
- grayscale
- invert
- opacity
- saturate
- sepia
- rgb
- rgba
- hsl
- hsla
- counter
- env
- minmax
- repeat
- hueRotate
- fitContent
- cubicBezier
- linearGradient
- radialGradient
- conicGradient
- repeatingConicGradient
- repeatingLinearGradient
- repeatingRadialGradient

## CSS Math

- add
- sub
- mul
- div
- prec