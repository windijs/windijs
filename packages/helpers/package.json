{
  "name": "@windijs/helpers",
  "version": "1.1.2",
  "description": "@windijs/helpers",
  "main": "dist/helpers.js",
  "module": "dist/helpers.mjs",
  "types": "dist/helpers.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/helpers.mjs",
      "require": "./dist/helpers.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "dependencies": {
    "@windijs/config": "1.0.7",
    "@windijs/shared": "1.0.5"
  },
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/helpers"
  }
}
