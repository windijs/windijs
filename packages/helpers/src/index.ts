// TODO: maybe add more helpers reference sass?
// https://sass-lang.com/documentation/modules

export * from "./types";
export * from "./build";
export * from "./config";
export * from "./color";
export * from "./common";
export * from "./css";
export * from "./data";
export * from "./dollar";
export * from "./escape";
export * from "./funcs";
export * from "./math";
export * from "./meta";
export * from "./namer";
export * from "./proxy";
export * from "./unit";
