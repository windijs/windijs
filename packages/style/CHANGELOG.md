# @windijs/style

## 1.1.0-beta.8

### Patch Changes

- Updated dependencies [[`da5e3466`](https://github.com/windijs/windijs/commit/da5e3466071e7c780b3875e177dad87e14495d3c), [`ba98151d`](https://github.com/windijs/windijs/commit/ba98151d7050fe6832999cf4c7ce0bec48d26fe0)]:
  - @windijs/helpers@1.1.0-beta.5
  - @windijs/core@1.1.0-beta.8

## 1.1.0-beta.7

### Patch Changes

- Updated dependencies [[`9940db66`](https://github.com/windijs/windijs/commit/9940db664965f9e7c04b8e6831cb035b79f2b212)]:
  - @windijs/helpers@1.1.0-beta.4
  - @windijs/core@1.1.0-beta.7

## 1.1.0-beta.6

### Patch Changes

- Updated dependencies []:
  - @windijs/core@1.1.0-beta.6

## 1.1.0-beta.5

### Patch Changes

- Updated dependencies [[`bbcd6f7f`](https://github.com/windijs/windijs/commit/bbcd6f7f355d015ed0e6a6183ce9204e5eb91892)]:
  - @windijs/core@1.1.0-beta.5

## 1.1.0-beta.4

### Patch Changes

- Updated dependencies [[`0b4d2f45`](https://github.com/windijs/windijs/commit/0b4d2f455a91b2c1eb09ac86e4f3783f36a86366)]:
  - @windijs/core@1.1.0-beta.4

## 1.1.0-beta.3

### Patch Changes

- [`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: remove circular devDependencies

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/core@1.1.0-beta.3
  - @windijs/helpers@1.1.0-beta.3
  - @windijs/shared@1.0.4-beta.1

## 1.1.0-beta.2

### Patch Changes

- [`602c375b`](https://github.com/windijs/windijs/commit/602c375bde2aeddc16399613d9bc45b42e3c5879) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: style handler type

- Updated dependencies [[`bd59c497`](https://github.com/windijs/windijs/commit/bd59c49748a68f189bfa8f529d8ba7d6bc0a22ae), [`d0d14195`](https://github.com/windijs/windijs/commit/d0d14195bea0f846097a9738da82278ca97468bc), [`6a71ee76`](https://github.com/windijs/windijs/commit/6a71ee76737a9309b31b06adc33a695461d1583d), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/core@1.1.0-beta.2
  - @windijs/helpers@1.1.0-beta.2
  - @windijs/shared@1.0.4-beta.0

## 1.1.0-beta.1

### Patch Changes

- [`8ab80baf`](https://github.com/windijs/windijs/commit/8ab80bafda9ab832d8d53f287e200f7fd497b7b6) Thanks [@voorjaar](https://github.com/voorjaar)! - chore: new versions

- Updated dependencies [[`8ab80baf`](https://github.com/windijs/windijs/commit/8ab80bafda9ab832d8d53f287e200f7fd497b7b6), [`e9a948cd`](https://github.com/windijs/windijs/commit/e9a948cd44d003a2c1e91ea2927658f59e3746dc), [`a03eabf8`](https://github.com/windijs/windijs/commit/a03eabf88e4ff96099bfc9a050a8440a500b0316)]:
  - @windijs/core@1.1.0-beta.1
  - @windijs/helpers@1.1.0-beta.1

## 1.1.0-beta.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

### Patch Changes

- Updated dependencies [[`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33)]:
  - @windijs/core@1.1.0-beta.0
  - @windijs/helpers@1.1.0-beta.0

## 1.0.5

### Patch Changes

- @windijs/core@1.0.5

## 1.0.4

### Patch Changes

- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3
  - @windijs/core@1.0.4
  - @windijs/helpers@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/core@1.0.3
  - @windijs/helpers@1.0.2
  - @windijs/shared@1.0.2

## 1.0.2

### Patch Changes

- @windijs/core@1.0.2

## 1.0.1

### Patch Changes

- chore: release v1.0.1
- Updated dependencies
  - @windijs/core@1.0.1
  - @windijs/helpers@1.0.1
  - @windijs/shared@1.0.1

## 1.0.0

### Patch Changes

- @windijs/core@1.0.0
- @windijs/helpers@1.0.0
