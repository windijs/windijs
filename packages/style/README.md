# @windijs/style

Vanilla style utilities and its type interface.