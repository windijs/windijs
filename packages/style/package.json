{
  "name": "@windijs/style",
  "version": "1.0.5",
  "description": "@windijs/style",
  "main": "dist/style.js",
  "module": "dist/style.mjs",
  "types": "dist/style.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    ".": {
      "import": "./dist/style.mjs",
      "require": "./dist/style.js"
    }
  },
  "dependencies": {
    "@windijs/core": "1.0.5",
    "@windijs/helpers": "1.0.3",
    "@windijs/shared": "1.0.3"
  },
  "devDependencies": {
    "@windijs/config": "1.0.4",
    "@windijs/colors": "1.0.2"
  },
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/style",
    "readmeFile": "./README.md"
  },
  "author": "Raven Satir",
  "license": "MIT"
}
