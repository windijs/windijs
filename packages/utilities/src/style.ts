import { createUtility, stylePropertyHandler } from "@windi/core";

export const style = createUtility("style").use(stylePropertyHandler()).init();
