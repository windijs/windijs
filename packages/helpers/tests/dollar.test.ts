import { $, build } from "@windijs/helpers";
import { style } from "@windijs/style";

// TODO: fix type bundler error

const utilities = [style.background.blue, style.padding.rem[4], style.borderRadius.px[4]];

test("dollar select HTML elements", () => {
  expect(build($.A(utilities))).toMatchSnapshot();

  expect(build($.Buttton(utilities))).toMatchSnapshot();

  expect(build($.Textarea(utilities))).toMatchSnapshot();
});

test("dollar with id", () => {
  expect(build($.ID("btn")(utilities))).toMatchSnapshot();
  expect(build($.ID.btn(utilities))).toMatchSnapshot();
  expect(build($.ID["red-button"](utilities))).toMatchSnapshot();
});

test("dollar with attribute", () => {
  expect(build($.ATTR.title(utilities))).toMatchSnapshot();
  expect(build($.ATTR("title")(utilities))).toMatchSnapshot();
  expect(build($.ATTR["some-attr"](utilities))).toMatchSnapshot();
  expect(build($.ATTR.href.match("https://example.org")(utilities))).toMatchSnapshot();
  expect(build($.ATTR.href.includes("example")(utilities))).toMatchSnapshot();
  expect(build($.ATTR.href.startsWith("https")(utilities))).toMatchSnapshot();
  expect(build($.ATTR.href.endsWith(".org")(utilities))).toMatchSnapshot();
  expect(build($.ATTR.class.contains("logo")(utilities))).toMatchSnapshot();
  expect(build($.ATTR.lang.hyphenMatch("zh")(utilities))).toMatchSnapshot();
});

test("dollar wiith all/root/host", () => {
  expect(build($.All(utilities))).toMatchSnapshot();
  expect(build($.Root(utilities))).toMatchSnapshot();
  expect(build($.Host(utilities))).toMatchSnapshot();
});

test("dollar with classes", () => {
  expect(build($.button(utilities))).toMatchSnapshot();
  expect(build($["bg-red-500"](utilities))).toMatchSnapshot();
});

test("dollar with css object", () => {
  expect(
    build(
      $.button({
        fontSize: "large",
        padding: "1rem",
        "&:hover": {
          background: "gray",
        },
      })
    )
  ).toMatchSnapshot();

  expect(
    build(
      $.ATTR.href.includes("example")({
        fontSize: "large",
        padding: "1rem",
        "&:hover": {
          background: "gray",
        },
      })
    )
  ).toMatchSnapshot();
});

test("dollar multi selectors", () => {
  expect(build($.button.$.A.$.Button(utilities))).toMatchSnapshot();
});

test("dollar nested selectors", () => {
  expect(build($.button.clicked(utilities))).toMatchSnapshot();
  expect(build($.ID["red-btn"].clicked(utilities))).toMatchSnapshot();
  expect(build($.button.clicked.$$.A(utilities))).toMatchSnapshot();
  expect(build($.button.clicked._.button.hovered(utilities))).toMatchSnapshot();
  expect(build($.button.clicked.__.A(utilities))).toMatchSnapshot();
  expect(build($.button.clicked._$_.A(utilities))).toMatchSnapshot();
  expect(build($.A.ATTR.href.endsWith(".org")(utilities))).toMatchSnapshot();
});

test("dollar call", () => {
  expect(build($(".test > a", utilities))).toMatchSnapshot();
  expect(build(Reflect.apply($, undefined, [".test > a", utilities]))).toMatchSnapshot();
});

test("dollar styles and exports", () => {
  $.init();

  const size = 100;

  $.circle(style.width.px[size], style.height.px[size], style.borderRadius.px[size * 0.5]);

  $.rounded(...$.circle.styles, style.fontSize.large);

  expect(build($.exports.map(i => i.style))).toMatchSnapshot();
});

test("extend last style", () => {
  $.init();

  const size = 100;

  $.circle(style.width.px[size], style.height.px[size], style.borderRadius.px[size * 0.5]);

  $.circle(style.backgroundColor.red);

  $.rounded(...$.circle.styles, style.fontSize.large);

  expect(build($.exports.map(i => i.style))).toMatchSnapshot();
});
