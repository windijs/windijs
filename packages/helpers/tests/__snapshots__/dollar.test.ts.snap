// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dollar call 1`] = `
".test > a {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar call 2`] = `
".test > a {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar multi selectors 1`] = `
".button, a, button {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 1`] = `
".button.clicked {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 2`] = `
"#red-btn.clicked {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 3`] = `
".button.clicked > a {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 4`] = `
".button.clicked .button.hovered {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 5`] = `
".button.clicked + a {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 6`] = `
".button.clicked ~ a {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar nested selectors 7`] = `
"a[href$=\\".org\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar select HTML elements 1`] = `
"a {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar select HTML elements 2`] = `
"buttton {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar select HTML elements 3`] = `
"textarea {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar styles and exports 1`] = `
".circle {
  width: 100px;
  height: 100px;
  border-radius: 50px;
}

.rounded {
  width: 100px;
  height: 100px;
  border-radius: 50px;
  font-size: large;
}"
`;

exports[`dollar wiith all/root/host 1`] = `
"* {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar wiith all/root/host 2`] = `
":root {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar wiith all/root/host 3`] = `
":host {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 1`] = `
"[title] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 2`] = `
"[title] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 3`] = `
"[some-attr] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 4`] = `
"[href=\\"https://example.org\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 5`] = `
"[href*=\\"example\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 6`] = `
"[href^=\\"https\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 7`] = `
"[href$=\\".org\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 8`] = `
"[class~=\\"logo\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with attribute 9`] = `
"[lang|=\\"zh\\"] {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with classes 1`] = `
".button {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with classes 2`] = `
".bg-red-500 {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with css object 1`] = `
".button {
  font-size: large;
  padding: 1rem;
}

.button:hover {
  background: gray;
}"
`;

exports[`dollar with css object 2`] = `
"[href*=\\"example\\"] {
  font-size: large;
  padding: 1rem;
}

[href*=\\"example\\"]:hover {
  background: gray;
}"
`;

exports[`dollar with id 1`] = `
"#btn {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with id 2`] = `
"#btn {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`dollar with id 3`] = `
"#red-button {
  background: blue;
  padding: 4rem;
  border-radius: 4px;
}"
`;

exports[`extend last style 1`] = `
".circle {
  width: 100px;
  height: 100px;
  border-radius: 50px;
}

.circle {
  background-color: red;
}

.rounded {
  background-color: red;
  font-size: large;
}"
`;
